agent/config/config.go

```diff
@@ -249,7 +252,9 @@ func environmentConfig() Config {
 	// invalid parse
 	// Blank is not a warning; we have sane defaults
 	if err != io.EOF && err != nil {
-		seelog.Warnf("Invalid format for \"ECS_RESERVED_PORTS_UDP\" environment variable; expected a JSON array like [1,2,3]. err %v", err)
+		msg := fmt.Sprintf("Invalid format for \"ECS_RESERVED_PORTS_UDP\" environment variable; expected a JSON array like [1,2,3]. err %v", err)
+		seelog.Warn(msg)
```

Comment:
This seemed the easiest way to construct a formatted string for use both in log output and in an error object returned to the caller.